src/import/prepare.rs

```diff
@@ -109,22 +108,51 @@ impl<Client: ImportClient> PrepareJob<Client> {
             }
 
             let range = RangeInfo::new(&start, k, ctx.raw_size());
-            if let Ok(true) = self.run_prepare_job(range) {
+            if let Ok(true) = self.run_prepare_job(range, &mut wait_scatter_regions) {
                 num_prepares += 1;
             }
 
             start = k.to_owned();
             ctx.reset(k);
         }
 
-        num_prepares
+        // We need to wait all regions for scattering finished.
+        let start = Instant::now();
+        while let Some(region_id) = wait_scatter_regions.pop() {
+            let start = Instant::now();
+            for i in 0..SCATTER_WAIT_MAX_RETRY_TIMES {
+                if self.client.is_scatter_region_finished(region_id)? {
+                    if i > 0 {
+                        debug!("waited between scatter"; "retry times" => %i, "takes" => ?start.elapsed());
+                    }
+                    info!("scatter region finished"; "region id" => %region_id, "takes" => ?start.elapsed());
+                    break;
+                } else if i == SCATTER_WAIT_MAX_RETRY_TIMES - 1 {
+                    warn!("scatter region still failed after exhausting all retries");
+                } else {
+                    // Exponential back-off with max wait duration
```

Comment:
seems that you should extract a util function or macro to reuse the retry logic.